contrib/go/src/python/pants/contrib/go/tasks/go_binary_fingerprint_strategy.py

```diff
@@ -0,0 +1,49 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import hashlib
+
+from pants.base.fingerprint_strategy import FingerprintStrategy
+from pants.contrib.go.targets.go_binary import GoBinary
+
+
+class GoBinaryFingerprintStrategy(FingerprintStrategy):
+  """Build flags aware fingerprint strategy.
+
+  This enables support for runtime merging of build flags (e.g.: config file, per-target, CLI),
+  which impact the output binary.
+  """
+
+  def __init__(self, build_flags_from_option, is_flagged, get_build_flags):
+    """
+    :param string build_flags_from_option: Runtime value of GoCompile build_flags option.
+    :param bool is_flagged: If build_flags was set via the command-line flag.
+    :param func get_build_flags: Function that merges build_flags from the various sources.
+    """
+    self._build_flags_from_option = build_flags_from_option
+    self._is_flagged = is_flagged
+    self._get_build_flags = get_build_flags
```

Comment:
This is all you care about; you're just a middleman for the 2 parameters above. How about currying the `GoCompile._get_build_flags` at the top of execute and using that for this strategy and in the main `GoCompile.execute` invalid target loop?

Reply:
I had a partial (ha!) implementation of that approach earlier, and thought this was more straightforward. But now that you bring it up too, yeah, that's probably the better approach. I'll update with this suggestion.